pydra/engine/core.py

```diff
@@ -534,28 +555,58 @@ def _run(self, rerun=False, **kwargs):
     def _collect_outputs(self, output_dir):
         run_output = self.output_
         output_klass = make_klass(self.output_spec)
-        output = output_klass(**{f.name: None for f in attr.fields(output_klass)})
+        output = output_klass(
+            **{f.name: attr.NOTHING for f in attr.fields(output_klass)}
+        )
         other_output = output.collect_additional_outputs(
             self.inputs, output_dir, run_output
         )
         return attr.evolve(output, **run_output, **other_output)
 
-    def split(self, splitter, overwrite=False, cont_dim=None, **kwargs):
+    def split(
+        self,
+        splitter: ty.Union[str, ty.List[str], ty.Tuple[str, ...], None] = None,
+        overwrite: bool = False,
+        cont_dim: ty.Optional[dict] = None,
+        **inputs,
+    ):
         """
         Run this task parametrically over lists of split inputs.
 
         Parameters
         ----------
-        splitter :
-            TODO
-        overwrite : :obj:`bool`
-            TODO
-        cont_dim : :obj:`dict`
+        splitter : str or list[str] or tuple[str] or None
+            the fields which to split over. If splitting over multiple fields, lists of
+            fields are interpreted as outer-products and tuples inner-products. If None,
+            then the fields to split are taken from the keyword-arg names.
+        overwrite : bool, optional
+            whether to overwrite an existing split on the node, by default False
+        cont_dim : dict, optional
             Container dimensions for specific inputs, used in the splitter.
             If input name is not in cont_dim, it is assumed that the input values has
             a container dimension of 1, so only the most outer dim will be used for splitting.
+        **split_inputs
+            fields to split over, will automatically be wrapped in a StateArray object
+            and passed to the node inputs
 
+        Returns
+        -------
+        self : TaskBase
+            a reference to the task
         """
+        if self._lzout:
+            raise Exception(
+                f"Cannot split {self} as its output interface has already been accessed"
+            )
```

Comment:
If you now try to split the node after you have accessed the lzout of the node, it will raise an error, e.g.

```python
wf.add(A(x=1, name="a"))
wf.add(B(w=wf.a.lzout.z))
wf.a.split(y=[1, 2, 3])  # <--- Will raise an error, because it will change z from a single value to a list/state-array
```